tokio/src/time/clock.rs

```diff
@@ -60,10 +60,52 @@ cfg_test_util! {
     #[derive(Debug)]
     struct Inner {
         /// Instant to use as the clock's base instant.
-        base: std::time::Instant,
+        base: LocalInstant,
 
         /// Instant at which the clock was last unfrozen
-        unfrozen: Option<std::time::Instant>,
+        unfrozen: Option<LocalInstant>,
+    }
+
+    #[derive(Debug, Copy, Clone)]
+    struct LocalInstant {
+        instant: std::time::Instant,
+        system: std::time::SystemTime,
+    }
+
+    #[derive(Debug, Copy, Clone)]
+    struct LocalDuration {
+        instant: Duration,
+        system: Duration,
+    }
+
+    impl LocalInstant {
+        fn now() -> Self {
+            Self {
+                instant: std::time::Instant::now(),
+                system: std::time::SystemTime::now(),
+            }
+        }
+
+        fn elapsed(&self) -> LocalDuration {
+            LocalDuration {
+                instant: self.instant.elapsed(),
+                system: self.system.elapsed().expect("Time went backwards!"),
```

Comment:
If I set the system time while running my tests, will this break even if I use only paused instants in my tests?

Reply:
What do you mean by set the system time? You mean via timectl or what ever? Not sure...but I would imagine that is out of scope of what we provide?